R/tm_g_patient_profile.R

```diff
@@ -426,89 +426,112 @@ srv_g_patient_profile <- function(id,
   checkmate::assert_class(data, "tdata")
 
   moduleServer(id, function(input, output, session) {
-    iv <- shinyvalidate::InputValidator$new()
-    iv$add_rule("sl_start_date", shinyvalidate::sv_required())
-    iv$add_rule("lb_var_show", shinyvalidate::sv_required())
-    iv$add_rule("ae_var", shinyvalidate::sv_required())
-    iv$enable()
-
     # only show the check box when domain data is available
-    observeEvent(ae_dataname, {
-      if (!is.na(ae_dataname)) {
-        output$select_ae_output <- renderUI({
-          checkboxInput(
-            session$ns("select_ae"),
-            "ADAE",
-            value = !is.na(ae_dataname)
-          )
-        })
-      }
+    checkboxes <- c(ex_dataname, ae_dataname, rs_dataname, lb_dataname, cm_dataname)
+    output$select_ADaM_output <- renderUI({
+      tagList(
+        helpText("Select", tags$code("ADaM"), "Domains"),
+        checkboxGroupInput(
+          inputId = session$ns("select_ADaM"),
+          label = NULL,
+          choices = checkboxes[!is.na(checkboxes)],
+          selected = checkboxes[!is.na(checkboxes)]
+        )
+      )
     })
 
-    observeEvent(ex_dataname, {
-      if (!is.na(ex_dataname)) {
-        output$select_ex_output <- renderUI({
-          checkboxInput(
-            session$ns("select_ex"),
-            "ADEX",
-            value = !is.na(ex_dataname)
-          )
-        })
-      }
+    select_plot <- reactive({
+      vapply(checkboxes, function(x) x %in% input$select_ADaM, logical(1L))
     })
 
-    observeEvent(rs_dataname, {
-      if (!is.na(rs_dataname)) {
-        output$select_rs_output <- renderUI({
-          checkboxInput(
-            session$ns("select_rs"),
-            "ADRS",
-            value = !is.na(rs_dataname)
-          )
-        })
-      }
-    })
+    observeEvent(
+      {
+        select_plot()
+        input$lb_var
+      },
+      {
+        req(select_plot()[lb_dataname])
+        req(input$lb_var)
+        ADLB <- data[[lb_dataname]]() # nolint
+        choices <- unique(ADLB[[input$lb_var]])
+        choices_selected <- if (length(choices) > 5) choices[1:5] else choices
 
-    observeEvent(cm_dataname, {
-      if (!is.na(cm_dataname)) {
-        output$select_cm_output <- renderUI({
-          checkboxInput(
-            session$ns("select_cm"),
-            "ADCM",
-            value = !is.na(cm_dataname)
-          )
-        })
+        updateSelectInput(
+          session,
+          "lb_var_show",
+          selected = choices_selected,
+          choices = choices
+        )
       }
-    })
+    )
```

Comment:
observe event on checkbox[lb_dataname] is unnecessary. We should set this observer only if lb_dataname is not empty

```suggestion
        if (!is.null(lb_dataname)) {
      observeEvent(
        input$lb_var,
        {
          req(input$lb_var)
          ADLB <- data[[lb_dataname]]() # nolint
          choices <- unique(ADLB[[input$lb_var]])
          choices_selected <- if (length(choices) > 5) choices[1:5] else choices

          updateSelectInput(
            session,
            "lb_var_show",
            selected = choices_selected,
            choices = choices
          )
        }
      )
    }
```